edx_django_utils/monitoring/internal/code_owner/middleware.py

```diff
@@ -0,0 +1,147 @@
+"""
+Middleware for code_owner custom attribute
+"""
+import logging
+
+from django.urls import resolve
+
+from ..transactions import get_current_transaction
+from ..utils import set_custom_attribute
+from .utils import get_code_owner_from_module, is_code_owner_mappings_configured
+
+log = logging.getLogger(__name__)
+
+
+class CodeOwnerMonitoringMiddleware:
+    """
+    Django middleware object to set custom attributes for the owner of each view.
+
+    For instructions on usage, see:
+    https://github.com/edx/edx-django-utils/blob/master/edx_django_utils/monitoring/docs/how_tos/add_code_owner_custom_attribute_to_an_ida.rst
+
+    Custom attributes set:
+    - code_owner: The owning team mapped to the current view.
+    - code_owner_mapping_error: If there are any errors when trying to perform the mapping.
+    - code_owner_path_error: The error mapping by path, if code_owner isn't found in other ways.
+    - code_owner_path_module: The __module__ of the view_func which was used to try to map to code_owner.
+        This can be used to find missing mappings.
+    - code_owner_transaction_error: The error mapping by transaction, if code_owner isn't found in other ways.
+    - code_owner_transaction_name: The current transaction name used to try to map to code_owner.
+        This can be used to find missing mappings.
+
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        self._set_code_owner_attribute(request)
+        return response
+
+    def process_exception(self, request, exception):
+        self._set_code_owner_attribute(request)
+
+    def _set_code_owner_attribute(self, request):
+        """
+        Sets the code_owner custom attribute, as well as several supporting custom attributes.
+
+        See CodeOwnerMonitoringMiddleware docstring for a complete list of attributes.
+
+        """
+        code_owner, path_error = self._set_code_owner_attribute_from_path(request)
+        if code_owner:
+            set_custom_attribute('code_owner', code_owner)
+            return
+        if not path_error:
+            # module found, but mapping wasn't configured
+            code_owner = self._set_code_owner_attribute_catch_all()
+            if code_owner:
```

Comment:
This code could definitely be refactored to be more clear, but I don't think it is invalid.  (In case it wasn't clear, this is not new code.  It is just being moved.)

As noted in the comment, the case is when the module is found in the request, but there wasn't a code owner configured for that path.  At that point, the only thing left to check is the catch all.  The transaction check lower down is for when the module wasn't found using the request.

I am going to be working in the code_owner code later (which is why I am doing this refactor), and I will consider looking into cleaning this up then.